cibuildwheel/architecture.py

```diff
@@ -46,19 +52,26 @@ def parse_config(config: str, platform: PlatformName) -> 'Set[Architecture]':
     def auto_archs(platform: PlatformName) -> 'Set[Architecture]':
         native_architecture = Architecture(platform_module.machine())
         result = {native_architecture}
+
         if platform == 'linux' and native_architecture == Architecture.x86_64:
             # x86_64 machines can run i686 docker containers
             result.add(Architecture.i686)
+
         if platform == 'windows' and native_architecture == Architecture.AMD64:
             result.add(Architecture.x86)
+
+        if platform == 'macos' and native_architecture == Architecture.arm64:
+            # arm64 can build and test both archs of a universal2 wheel.
+            result.add(Architecture.universal2)
+
         return result
 
     @staticmethod
     def all_archs(platform: PlatformName) -> 'Set[Architecture]':
         if platform == 'linux':
             return {Architecture.x86_64, Architecture.i686, Architecture.aarch64, Architecture.ppc64le, Architecture.s390x}
         elif platform == 'macos':
-            return {Architecture.x86_64}
+            return {Architecture.x86_64, Architecture.arm64, Architecture.universal2}
```

Comment:
I don't know... my gut is that it shouldn't depend on macOS version, because the new archs also require CPython 3.9, so we should localise that logic in `get_python_configurations`, not spread it out over the codebase.

Reply:
Especially since this currently only affects 'all', then I think it's okay. Now if we added `universal2` to `auto`, then this be a (much) bigger issue, as `auto` should not break on macOS < 11.

Reply:
Actually the requirement is Xcode 12, so it's not something we should be fiddling with here.